harness/determined/keras/_tf_keras_trial.py

```diff
@@ -567,7 +567,7 @@ def _save_checkpoint(self, path: pathlib.Path) -> workload.Response:
 
         # Save RNG state.
         rng_state = {"np_rng_state": np.random.get_state(), "random_rng_state": random.getstate()}
-        if version.parse(tf.__version__) < version.parse("2.0.0"):
+        if not (version.parse(tf.__version__) >= version.parse("2.0.0") and tf._tf2.enabled()):
```

Comment:
Why convert the  `<` into a `not >=`, out of curiosity?

Reply:
it may be matter of taste, but in my opinion, `NOT(version >= 2 AND version2-specific-condition)` is more readable, because inside of NOT you have a standard pattern of checking for something specific for the new version. 
This makes use of short circuiting (otherwise `tf._tf2` will error out), and short circuiting based on OR's may be harder to read fast.